src/peft/optimizers/loraplus.py

```diff
@@ -0,0 +1,98 @@
+"""
+This module contains the implementation of the LoraPlus optimizer.
+"""
+
+from operator import attrgetter
+
+import torch.nn as nn
+from torch.optim import Optimizer
+from transformers.pytorch_utils import ALL_LAYERNORM_LAYERS
+from transformers.trainer_pt_utils import get_parameter_names
+
+from ..peft_model import PeftModel
+from ..tuners.lora.layer import Embedding
+
+
+def create_loraplus_optimizer(
+    model: PeftModel,
+    optimizer_cls: type[Optimizer],
```

Comment:
`type[Optimizer]` fails with Python 3.8, please add the `__future__` import to this file as well.